### ISSUE TITLE
Regex columnizer not compatible with Time Sync?

### ISSUE THREAD

**Issue by lekjart** [zarunbal/LogExpert#151]
I modified Regex1 to a regex that split my specific type of log into Date, Time and Message. The columnizer works fine, but the Time Sync option is not available when using it. Is the Time Sync only available when using the Timestamp Columnizer/?

**Comment by lekjart** [comment]
Here is the regex, just in case I am missing anything there:
\[(?<Date>\d{4}.\d{2}.\d{2})-(?<Time>\d{2}.\d{2}.\d{2}:\d{3})](?<Message>.*)

**Comment by Hirogen** [comment]
> Timestamp Columnizer
Yes, because, logexpert does not know, which part of your regex the "timestamp" is.

**Comment by lekjart** [comment]
Ok, I thought that by naming it Date and Time in the regex itself would do the trick, as that at least created columns that had the same names as other columnizers that support Time Sync. Is there any way to do it differently in the regex in order for this to work?

**Comment by Hirogen** [comment]
> Ok, I thought that by naming it Date and Time in the regex itself would do the trick, as that at least created columns that had the same names as other columnizers that support Time Sync. Is there any way to do it differently in the regex in order for this to work?

I don't think so, the other columnizers are coded as C# so there is more flexibility in what is shown where, with regex, you have only the catching groups that you create as columns.

I'm not sure, there might be the possibility to read the keyword date and "link" it to the time-sync, but the investment to write something universal is probably quite high.

**Comment by zarunbal** [comment]
At the moment its not implement, but at my current view this should be possible to implement, to support Time Sync with Regex columnizer

**Comment by RandallFlagg** [comment]
@Hirogen please explain what exactly is required to be done

**Comment by Hirogen** [comment]
The problem is, you need to specify the part of the regex that is the "time/date" component

**Comment by RandallFlagg** [comment]
I still don't understand. Is this a feature of the regex columnizer that is mising? Can you explain what needs to be done with an example of how and where and anything else you think might help me to understand the issue?

**Comment by Hirogen** [comment]
yes its a missing feature, because you need to configure or be able to specify what part of the given regex is the "date-time" part and that is quite complex. We could enforce Named Groups, but thats a different beast.
